### PR TITLE
Migrate to JDK 11 and Java 11 syntax [part 2]

### PR DESCRIPTION
**Pull request by armiol**
This PR is the follow-up to #1425.

The main focus is the `var` introduction to the part of `server` module packages.

Some of the existing warnings were addressed as well.

The library version is set to `2.0.0-SNAPSHOT.83`.

### CODE REVIEW

File: server/src/main/java/io/spine/server/aggregate/model/PartFactory.java [hunk 1]
```diff
@@ -104,16 +104,16 @@ public A create(Object root) {
...
+        var result = (Constructor<A>) ctor;
         return result;
     }
 
     private static ModelError noSuchConstructor(Class<?> partClass, Class<?> rootClass) {
-        String errMsg =
+        var errMsg =
                 format("`%s` class must declare a constructor with one parameter of the `%s` type.",
                        partClass.getName(),
                        rootClass.getName());
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/aggregate/model/PartFactory.java:115]:
```suggestion
        var errMsg = format(
                "`%s` class must declare a constructor with one parameter of the `%s` type.",
                partClass.getName(),
                rootClass.getName()
        );
```

File: server/src/main/java/io/spine/server/delivery/Conveyor.java [hunk 2]
```diff
@@ -244,11 +238,10 @@ void keepForLonger(InboxMessage message, Duration howLongTooKeep) {
...
      */
     void flushTo(InboxStorage storage) {
-        List<InboxMessage> dirtyMessages =
-                messages.values()
-                        .stream()
-                        .filter(message -> this.dirtyMessages.contains(message.getId()))
-                        .collect(toList());
+        var dirtyMessages = messages.values()
+                .stream()
+                .filter(message -> this.dirtyMessages.contains(message.getId()))
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/delivery/Conveyor.java:243]:
```suggestion
                .filter(message -> dirtyMessages.contains(message.getId()))
```

File: server/src/main/java/io/spine/server/enrich/EnricherBuilder.java [hunk 3]
```diff
@@ -204,7 +203,7 @@ public boolean equals(Object obj) {
             if (!(obj instanceof Key)) {
                 return false;
             }
-            final Key other = (Key) obj;
+            final var other = (Key) obj;
             return Objects.equals(this.sourceClass, other.sourceClass)
                     && Objects.equals(this.enrichmentClass, other.enrichmentClass);
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/enrich/EnricherBuilder.java:208]: I know this is out of scope for this PR, but can we refer to the class FQNs? `Class.equals(..)` works via `==` and thus is class loader-dependent.

**Review by dmdashenkov — APPROVED** [review]
LGTM with a couple of suggestions.

File: server/src/main/java/io/spine/server/enrich/SchemaFn.java [hunk 4]
```diff
@@ -83,11 +82,12 @@ static void put(Container.Builder container, Message output) {
...
                      Message sourceMessage,
                      EnrichableMessageContext context,
-                     EnrichmentFn function) {
-        checkNotNull(
+                     EnrichmentFn<?, ?, ?> function) {
+        requireNonNull(
                 output,
-                "`%s` produced `null` for the source message `%s` (context: `%s`).",
-                function, sourceMessage, context
+                String.format(
```
Comment (alexander-yevsyukov) [inline server/src/main/java/io/spine/server/enrich/SchemaFn.java:88]: Please static-import `format()`, if possible.

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM, with few minor requests to consider before merging.

**Pull request merged**